src/routes/Player/usePlayer.js

```diff
@@ -110,7 +110,19 @@ const usePlayer = (urlParams) => {
         }, 'player');
     }, []);
     const player = useModelState({ model: 'player', action, init, map });
-    return [player, updateLibraryItemState, pushToLibrary];
+    const playlist = React.useMemo(() => {
+        if (player.selected === null || typeof player.selected.stream.url !== 'string') {
+            return null;
+        }
+
+        const playlist = `#EXTM3U\n\n#EXTINF:0,${player.title}\n${player.selected.stream.url}`;
```

Comment:
player.title must be wrapped in encodeURIComponent
player.selected.stream.url must be wrapped in encodeURI
otherwise if there are some specific characters btoa will throw an exception